### DISCUSSION TITLE
JSON parse error when Shared data "other server" connection is lost

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1060] [Fixed Issues]
Running rAudio 20221010

If shared data client or server is active AND the shared data server connection is lost (network down, turned off, etc.) , the following will occur:
1) The GUI will show a flashing rAudio logo if viewed from my iPhone or
2) The following error will eventually be displayed if viewed from a PC web browser (Firefox in my case):

![JSON parse error](https://user-images.githubusercontent.com/76575081/195098455-754fb11d-d620-4fc0-b71c-16ab0240b703.jpg)

This behavior did NOT occur after disabling Shared data and then removing the NAS connection.

I originally reported this in #1019  (which I have since closed and re-issue in this bug report).

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by Sagnasty** [comment]
Test results with rAudio 202210131 update:

Ensured that shared data (client) was enabled.
Applied 202210131  update.
Turned off NAS NFS service.
Attempted to access webradios on shared data client machine.
Server offline message was displayed:
![image](https://user-images.githubusercontent.com/76575081/195610268-36e1c006-5dc4-4b54-b6e2-c48e305e3ec6.png)
clicked OK to disable shared data.
Received JSON error (PC/firefox):
![image](https://user-images.githubusercontent.com/76575081/195610632-2d4b291e-7ba6-4f6c-a25d-948d61918b6c.png)

Re-establishing NAS NFS service and then cycling power to machine resulted in the same JSON error.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Reply by rern** [reply]
The OK button runs the same script as of the switch to disable Shared Data.
Can you disable with the switch?
If no, post error output of the script:
```sh
system.sh shareddatadisconnect
```

**Reply by Sagnasty** [reply]
Yes, I can disable from shared data switch.

I also tried this again from the Server offline message and this time it worked without issue (I gave a bit more time in between disabling NAS NFS service and then attempting to access the shared library which is a more realistic test case).   

I noticed that all 4 radio paradise webradios were available and functional after disabling shared data.
is it possible to also include the shared data webradios stations (local copy to machine)?
Then it would be possible to at least play the users web radio stations while shared data is disabled / not available.
